modopt/opt/proximity.py

```diff
@@ -1042,3 +1042,69 @@ def _cost_method(self, *args, **kwargs):
             print(' - K-SUPPORT NORM (X):', cost_val)
 
         return cost_val
+
+
+class GroupLASSO(ProximityParent):
+    """This class implements the proximity operator of the group-lasso
+    regularization, with groups dimension being the first dimension
+    Attributes:
+    ----------
+    weights : np.ndarray
+        Input array of weights
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from modopt.opt.proximity import GroupLASSO
+    >>> A = np.arange(15).reshape(3,5)
+    array([[ 0,  1,  2,  3,  4],
+           [ 5,  6,  7,  8,  9],
+           [10, 11, 12, 13, 14]])
+    >>> prox_op = GroupLASSO(weights=3)
+    >>> prox_op.op(A)
+    array([[ 0.        ,  0.76133281,  1.5725177 ,  2.42145809,  3.29895251],
+           [ 3.65835921,  4.56799689,  5.50381195,  6.45722157,  7.42264316],
+           [ 7.31671843,  8.37466096,  9.4351062 , 10.49298505, 11.5463338 ]])
+    >>> prox_op.cost(A, verbose=True)
+         211.37821733946427
+    """
+    def __init__(self, weights):
+        self.weights = weights
+        self.op = self._op_method
+        self.cost = self._cost_method
+
+    def _op_method(self, data, extra_factor=1.0):
+        """ Operator
+        This method returns the input data thresholded by the weights
+
+        Parameters
+        ----------
+
+        data : DictionaryBase
+            Input data array
+        extra_factor : float
+            Additional multiplication factor
+
+        Returns
+        -------
+        DictionaryBase thresholded data
+        """
+        norm_2 = np.linalg.norm(data, axis=0)
+        return data * np.maximum(0, 1.0 - self.weights * extra_factor /
+                                 np.maximum(norm_2, np.finfo(np.float32).eps))
+
+    def _cost_method(self, data):
+        """Cost function
+        This method calculate the cost function of the proximable part.
+
+        Parameters
+        ----------
+
```

Comment:
Remove blank line

Reply:
I thought that was needed for documentation?

Reply:
It might not make any difference, but all of the code should be consistent, so either everything with no space (as is the current standard, which seems to work fine) or everything with a space, but certainly not a mix of the two.